core/posts/views.py

```diff
@@ -145,17 +147,35 @@ def retrieve(self, request, pk):
                 "message": "You are not authorized to view this post.",
                 "query": "post"
             }, status=status.HTTP_403_FORBIDDEN)
-        images = Posts.objects.filter(post_id=post.post_id).exclude(contentType__contains="text")
         serializer = PostsSerializer(post, context={'request': request})
-        img_serializer = PostsSerializer(images, many=True, context={'request': request})
         return Response({
             "query": "posts",
             "count": 1,
             "size": 1,
-            "posts": [serializer.data],
-            "images": img_serializer.data
+            "posts": [serializer.data]
         })
 
+    @action(detail=True, url_path='image', methods=["GET"])
+    def image(self, request, pk):
+        try:
+            post = Posts.objects.get(pk=pk)
+        except:
+            return Response({
+                "success": False,
+                "message": "No post was found with that ID",
+                "query": "getImage"
+            }, status=404)
+        if not can_user_view(request.user, post):
+            return Response({
+                "success": False,
+                "message": "You are not authorized to view this post.",
+                "query": "post"
+            }, status=status.HTTP_403_FORBIDDEN)
+        data = post.content.split(",")[1]
+        data = data.encode()
+        data = base64.b64decode(data)
+        return HttpResponse(data, content_type="image/png")
```

Comment:
This should be able to handle jpeg images as well

Reply:
my bad